compiler/rustc_ast/src/token.rs

```diff
@@ -602,6 +602,15 @@ impl Token {
         self.is_non_raw_ident_where(|id| id.name == kw)
     }
 
+    /// Returns `true` if the token is a given keyword, `kw` or if `case_insensitive` is true and this token is an identifier equal to `kw` ignoring the case.
+    pub fn is_keyword_case(&self, kw: Symbol, case_insensitive: bool) -> bool {
```

Comment:
In general we want to avoid using `bool` and instead introduce an enum to have more clarity in call sites and a place to hang documentation off of (but for functions with a single arg it isn't as big a deal).

Reply:
Replaced with an enum 👍🏻